### PR TITLE
Compound external name handling in Mongo SQL Connector [HZ-2260]

### PR DESCRIPTION
**Pull request by TomaszGaweda**
External name can be:
`database.collection`
or just `collection`.

In the second case, the database must be provided in the data link definition.

Jira: [#HZ-2260](https://hazelcast.atlassian.net/browse/HZ-2260)

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by ldziedziul — APPROVED** [review]

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java [hunk 1]
```diff
@@ -84,21 +84,21 @@ public void readsFromMongo(boolean includeIdInMapping, boolean forceTwoSteps) {
         collection.insertOne(new Document("firstName", "Anakin").append("lastName", "Skywalker").append("jedi", true));
         collection.insertOne(new Document("firstName", "Rey").append("jedi", true));
 
-        execute("CREATE MAPPING " + collectionName
-                + " ("
+        String mappingName = collectionName;
+        String externalName = databaseName + "." + mappingName;
+        execute("CREATE MAPPING " + mappingName + " external name " + externalName + " ("
```
Comment (frant-hartm) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java:89]:
```suggestion
        execute("CREATE MAPPING " + mappingName + " EXTERNAL NAME " + externalName + " ("
```

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java [hunk 2]
```diff
@@ -84,21 +84,21 @@ public void readsFromMongo(boolean includeIdInMapping, boolean forceTwoSteps) {
...
                 + (includeIdInMapping ? " id VARCHAR external name _id, " : "")
```
Comment (frant-hartm) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java:90]:
```suggestion
                + (includeIdInMapping ? " id VARCHAR EXTERNAL NAME _id, " : "")
```

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/FieldResolver.java [hunk 3]
```diff
@@ -160,16 +159,23 @@ private void validateType(MappingField field, DocumentField documentField) {
...
+    }
+
     Map<String, DocumentField> readFields(String[] externalNames,
                                           String dataConnectionName,
                                           Map<String, String> options,
                                           boolean stream) {
-        String collectionName = externalNames[0]; // TODO HZ-2260
+        String collectionName = externalNames.length == 2 ? externalNames[1] : externalNames[0];
+        String databaseName = Options.getDatabaseName(nodeEngine, externalNames, dataConnectionName);
+        ClientWithDataConnection connect = connect(dataConnectionName, options);
```
Comment (frant-hartm) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/FieldResolver.java:174]: You actually don't need both client and dataConnection. It's enough in `connect` to do the following:

- get dataConnection 
- get client
- release the dataConnection
- close the client

You don't need to keep the dataConnection until you close the client.

**Review by frant-hartm — CHANGES_REQUESTED** [review]

**Comment by TomaszGaweda** [comment]
all done @frant-hartm :)

**Review by frant-hartm — APPROVED** [review]

**Comment by TomaszGaweda** [comment]
run-lab-run

**Pull request merged**